### PR TITLE
69-stock-management

### PR DESCRIPTION
**Pull request by lordofdankness**
https://github.com/indera-solutions/nima/issues/69

### CODE REVIEW

File: apps/api/src/order/order.service.ts [hunk 1]
```diff
@@ -315,4 +318,9 @@ export class OrderService {
 		await this.orderRepository.deleteById(id);
 		return res;
 	}
+
+	private async returnProductVariantStock(orderLines: OrderLineDto[]): Promise<void> {
```
Comment (Schavras) [inline apps/api/src/order/order.service.ts:322]: na kaleite kai sto `RETURNED`

File: apps/api/src/products/product-variant.service.ts [hunk 2]
```diff
@@ -329,4 +329,18 @@ export class ProductVariantService {
 		}
 		return roundToDigit(lowestPrice);
 	}
+
+	async checkStock(params: { productVariantId: number }): Promise<Pick<ProductVariantEntity, 'stock' | 'trackInventory'>> {
```
Comment (Schavras) [inline apps/api/src/products/product-variant.service.ts:333]: afto den kaleite kapou?

**Review by Schavras — CHANGES_REQUESTED** [review]

File: apps/api/src/order/order.service.ts (hunk 1 above)
Comment (lordofdankness) [reply apps/api/src/order/order.service.ts:322]: o logos pou den to evala sto return einai giati skeftika an to return einai gia defective product theloume an mpei apefthias sto stock? An den mas noiazei to vazw.

File: apps/api/src/products/product-variant.service.ts (hunk 2 above)
Comment (lordofdankness) [reply apps/api/src/products/product-variant.service.ts:333]: to eixame sto palio kai to evala na testarw kati kai eipa na to afisw mporeis na xreiasti, na to vgalw?

File: apps/api/src/order/order.service.ts (hunk 1 above)
Comment (Schavras) [reply apps/api/src/order/order.service.ts:322]: vale ta na girnaei, giati mporei apla na thelei allo noumero papoutsi, xwris na exei kapoio provlima

File: apps/api/src/products/product-variant.service.ts (hunk 2 above)
Comment (Schavras) [reply apps/api/src/products/product-variant.service.ts:333]: nai, exw grapsei sto issue se poia simeia prepei na to kaleis

**Pull request merged**